setup.py

```diff
@@ -33,6 +33,7 @@
         "python-terraform>=0.10.0,<0.11.0",
         "boto3>=1.17.49,<=1.18.0",
         "botocore>=1.20.49,<=1.21.0",
+        "boto3-stubs[ec2,s3,rds]~=1.18",
```

Comment:
This can go in [requirements-type.txt](https://github.com/app-sre/qontract-reconcile/blob/master/requirements-type.txt) because it's only needed during type checking.

Reply:
It looks like `types-boto3` is already there, which is a proxy package for `boto3-stubs`. I wonder if that might have been related in some way. In any case, I would try removing `types-boto3` and putting this in its place, then seeing if you can reproduce. Stub and type packages shouldn't be imported directly, so shouldn't need to be runtime dependencies.

Reply:
same errors during unit tests, looking like:
```
________ ERROR collecting reconcile/test/test_terraform_vpc_peerings.py ________
ImportError while importing test module '/package/reconcile/test/test_terraform_vpc_peerings.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
reconcile/test/test_terraform_vpc_peerings.py:4: in <module>
    import reconcile.terraform_vpc_peerings as integ
reconcile/terraform_vpc_peerings.py:6: in <module>
    import reconcile.utils.aws_api as awsapi
reconcile/utils/aws_api.py:14: in <module>
    from mypy_boto3_ec2 import EC2Client
E   ModuleNotFoundError: No module named 'mypy_boto3_ec2'
```
I am keeping it in `setup.py` for now, until a better / proper way is found

Reply:
I see the issue now. It wasn't loading that file for me due to a large number of changes. You are importing the type library, which is unique to boto3 due to the dynamic nature of that library.

This is covered in the [docs](https://pypi.org/project/boto3-stubs/) under `Pylint compatibility` section. A conditional with `typing.TYPE_CHECKING` is traditionally how you'd import only for type checking. The only thing that might be better than what they've described is representing the types as strings (ex. `my_func() -> "EC2Client"`). The string version can be used when you're referring to a type before it is defined (think self reference in classes), but might also be useful here. I believe that'd work with 3.6, but what they've described in the docs would work too (although it's a bit less nice).

Reply:
Also, the string representation and forward references are covered here: https://www.python.org/dev/peps/pep-0484/#forward-references

This won't be required in Python 3.7+, but for now it's fine.